### PR TITLE
DOC: stats: reorganize hypothesis tests in main page

### PR DESCRIPTION
**Pull request by mdhaber**
Reorganize hypothesis tests section.

Rather than looking at the diff, it's probably easier to glance at what was moved out of other sections, then just read the "Hypothesis Tests and related functions" section of the modified file.

@tupui @tirthasheshpatel @chrisb83 I thought I'd get your opinion before submitting the PR upstream.

### CODE REVIEW

**Comment by tupui** [comment]
Definitely wanted 👍

**Comment by chrisb83** [comment]
Agree, this is a good idea. The current overview of stats functions is hard to follow

**Comment by mdhaber** [comment]
Thanks. @chrisb83 when you added section headings in  https://github.com/scipy/scipy/pull/9062, f_oneway ended up as a "correlation function". Was that just an artifact of the way the functions were already ordered? Isn't it really an independent sample test?

**Comment by chrisb83** [comment]
Agree, f_oneway does not belong to correlation functions

**Comment by mdhaber** [comment]
Thanks! If you'd take a look at the PR to the main repo (gh-17767 - where to put `kstest`?) I'd appreciate it!
